### PR TITLE
Create dataset from Postgres (continue)

### PR DESCRIPTION
**Pull request by matthewma7**
This PR is the continuation of https://github.com/Kitware/minerva/pull/410.
The original PR was accidentally merged and closed. Its changes have been reverted.

### CODE REVIEW

File: server/utility/minerva_utility.py [hunk 1]
```diff
@@ -48,6 +48,11 @@ def findMinervaFolder(currentUser, user, create=False):
                            PluginSettings.MINERVA_FOLDER, create)
 
 
+def findPublicFolder(currentUser, user, create=False):
```
Comment (dorukozturk) [inline server/utility/minerva_utility.py]: My editor was changing the file modes and it also happened here. Can you change the file mode back to 100644? Or I can do it.

File: server/rest/postgres_geojson.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+                        'table': table,
+                        'schema': schema}],
+            'fields': fields,
+            'group': group,
+            'filters': filters,
+            'limit': -1,
+            'format': output_format
+        }
+
+    # def _getProperties(self, params):
```
Comment (dorukozturk) [inline server/rest/postgres_geojson.py]: Can we remove this if not needed.

File: server/utility/minerva_utility.py (hunk 1 above)
Comment (matthewma7) [inline server/utility/minerva_utility.py]: Oh, I probably unintentionally changed it. If it's not too much trouble, you could change it back.

Comment (dorukozturk) [inline server/utility/minerva_utility.py]: Ok I reverted it back.

File: server/rest/postgres_geojson.py [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+
+        if geometryField['type'] == 'built-in':
+            buildInGeomField = geometryField['field']
+            properties = [field, {
+                'func': aggregateFunction,
+                'param': {'field': field}
+            }]
+            # add string fields with concat aggregate function and in the format for
+            # json_build_object
+            for property in [(i['name'], {
```
Comment (dorukozturk) [inline server/rest/postgres_geojson.py]: Can we make this more readable? Maybe extract some of the expressions to separate variables?

Comment (matthewma7) [inline server/rest/postgres_geojson.py]: Totally, I got confused with the weird indentation myself when I read it just then.

File: server/rest/postgres_geojson.py [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+            group = [buildInGeomField]
+        elif geometryField['type'] == 'link':
+            fields = [{
+                'func': aggregateFunction,
+                'param': {'field': field},
+                'reference': field
+            }]
+            group = [x['value'] for x in geometryField['links']]
+            # add string fields with concat aggregate function
+            for i in [i for i in self._getColumns(assetstore, {'table': params['table']})
```
Comment (dorukozturk) [inline server/rest/postgres_geojson.py]: Here you are looping through a list that is created by a list comprehension. I think you don't need the list comprehension here. IMHO you should either do a for loop or list comprehension but not both. I believe this is too complicated to be a list comprehension and will be easier to read if it is just a for loop.

Comment (matthewma7) [inline server/rest/postgres_geojson.py]: Agreed. Updated.

File: web_external/js/views/widgets/PostgresWidget.js [hunk 5]
```diff
@@ -0,0 +1,473 @@
...
+        'change select#geometry-built-in-field': '_geometryBuiltInFieldChange',
+        'change .geometry-field-type': '_geometryFieldTypeChange',
+        'change select.link-target': '_linkTargetChange',
+        'change select.link-field': '_linkFieldChange',
+        'change select.link-operator': '_linkOperatorChange',
+        'change select.link-value,input.link-value': '_linkValueChange',
+        'click button.add-link': '_addLinkClick',
+        'click button.remove-link': '_removeLinkClick'
+    },
+    formToTable: {
```
Comment (manthey) [inline web_external/js/views/widgets/PostgresWidget.js]: Is this used for anything?

Comment (matthewma7) [inline web_external/js/views/widgets/PostgresWidget.js]: Good catch. No, it's not used. I'll remove it.

**Pull request merged**